vantage6/cli/rabbitmq/queue_manager.py

```diff
@@ -0,0 +1,222 @@
+        """
+        # get volumes which contain rabbitMQ configuration and a location to
+        # save rabbitMQ files so that they persist after container is stopped
+        volumes = self._get_volumes()
+
+        # expose port 5672 inside the container as port 5672 on the host, and
+        # same for 15672 in container to 8080 on host
+        # TODO check if these ports are not already used on the host
+        ports = {
+            f'{QUEUE_PORT}/tcp': QUEUE_PORT,
+            # TODO this is for the management tool, do we keep this? Not used
+            # at the moment..
+            '15672/tcp': 8080
+        }
+
+        # if a RabbitMQ container is already running, kill and remove it
+        remove_container_if_exists(
+            docker_client=self.docker, name=self.rabbit_container_name
+        )
+
+        # run rabbitMQ container
+        self.rabbit_container = self.docker.containers.run(
+            name=self.rabbit_container_name,
+            image=self.image,
+            volumes=volumes,
+            ports=ports,
+            detach=True,
+            restart_policy={"Name": "always"},
+            hostname=f'{APPNAME}-{self.ctx.name}-rabbit',
+            labels={
+                f"{APPNAME}-type": "rabbitmq",
+            }
+        )
+
+        # Wait until RabbitMQ is up before continuing with other stuff
+        self._wait_for_startup()
+
+    def _wait_for_startup(self) -> None:
+        """ Wait until RabbitMQ has been initialized """
+        # TODO make the time / # attempts settable? Is this enough time (5m)?
+        ATTEMPTS = 31
+        is_running = False
+        for _ in range(ATTEMPTS):
+            if self.is_running():
+                is_running = True
+                break
+            debug("RabbitMQ is not yet running, trying again in 10s...")
+            time.sleep(10)
+        if is_running:
+            info("RabbitMQ was started successfully!")
+        else:
+            # TODO exit or disable?
+            error("Could not start RabbitMQ! Exiting...")
+            exit(0)
+
+    def is_running(self) -> bool:
+        """
+        Returns
+        -------
+        bool
+            Whether the container has fully initialized RabbitMQ or not
+        """
+        response = self.rabbit_container.exec_run(
+            cmd="rabbitmqctl status --formatter json"
+        )
+        return response.exit_code == 0
+
+    def _get_volumes(self) -> Dict:
+        """
+        Prepare the volumes for the RabbitMQ container. The RabbitMQ should
+        set up the right vhost and users to allow the server to communicate
+        with RabbitMQ as configured.
+        """
+        # default RabbitMQ configuration: replace the user/password with the
+        # credentials from the configuraiton
+        rabbit_definitions = self._get_rabbitmq_definitions()
+
+        # write the RabbitMQ definition to file(s)
+        definitions_filepath = Path(self.ctx.data_dir / 'definitions.json')
```

Comment:
I think it would be good to make this a module constant, so we only need to change it at one spot